src/connect.rs

```diff
@@ -41,19 +41,31 @@ enum Inner {
 
 impl Connector {
     #[cfg(not(feature = "tls"))]
-    pub(crate) fn new(proxies: Arc<Vec<Proxy>>) -> ::Result<Connector> {
+    pub(crate) fn new<T>(proxies: Arc<Vec<Proxy>>, local_addr: T) -> ::Result<Connector>
+    where
+        T: Into<Option<IpAddr>>
+    {
+
         let http = http_connector()?;
+        http.set_local_address(local_addr);
         Ok(Connector {
             proxies,
             inner: Inner::Http(http)
         })
     }
 
     #[cfg(feature = "default-tls")]
-    pub(crate) fn new_default_tls(tls: TlsConnectorBuilder, proxies: Arc<Vec<Proxy>>) -> ::Result<Connector> {
+    pub(crate) fn new_default_tls<T>(
+        tls: TlsConnectorBuilder,
+        proxies: Arc<Vec<Proxy>>,
+        local_addr: T) -> ::Result<Connector>
+        where
+            T: Into<Option<IpAddr>>,
+    {
         let tls = try_!(tls.build());
 
         let mut http = http_connector()?;
+        http.set_local_address(local_addr.into());
```

Comment:
@seanmonstar  Why is this the only place I had to use into()? The error was  something along the lines of `expecting std::option::Option, got type parameter`

Reply:
Actually looks like some of the builds failed with that same error. I will try adding `into()` to the rest.